### PR TITLE
Add CMEK E2E Test

### PR DESCRIPTION
**Pull request by davidz627**
/assign @saad-ali @msau42 

A CMEK E2E test that provisions with CMEK, tests lifecycle, revokes key, makes sure disk usage fails, activates key, makes sure lifecycle succeeds again.

### CODE REVIEW

File: test/e2e/tests/single_zone_e2e_test.go [hunk 1]
```diff
@@ -294,6 +302,165 @@ var _ = Describe("GCE PD CSI Driver", func() {
...
+			CryptoKeyId: keyId,
+			CryptoKey: &kmspb.CryptoKey{
+				Purpose: kmspb.CryptoKey_ENCRYPT_DECRYPT,
+				VersionTemplate: &kmspb.CryptoKeyVersionTemplate{
+					Algorithm: kmspb.CryptoKeyVersion_GOOGLE_SYMMETRIC_ENCRYPTION,
+				},
+			},
+		}
+		key, err := kmsClient.CreateCryptoKey(ctx, keyReq)
+		if !gce.IsGCEError(err, "alreadyExists") {
```
Comment (msau42) [inline test/e2e/tests/single_zone_e2e_test.go]: should it already exist if we're generating uuid every time?

File: test/e2e/tests/single_zone_e2e_test.go [hunk 2]
```diff
@@ -294,6 +302,165 @@ var _ = Describe("GCE PD CSI Driver", func() {
...
+
+		controllerInstance := testContext.Instance
+		controllerClient := testContext.Client
+
+		p, z, _ := controllerInstance.GetIdentity()
+		locationID := "global"
+
+		// The resource name of the key rings.
+		parentName := fmt.Sprintf("projects/%s/locations/%s", p, locationID)
+		keyRingId := "gce-pd-csi-test-ring"
```
Comment (msau42) [inline test/e2e/tests/single_zone_e2e_test.go]: Is there any possible issue of multiple tests sharing the same key ring?  Or leaving a key ring around at the end of the test?

File: test/e2e/tests/single_zone_e2e_test.go [hunk 3]
```diff
@@ -294,6 +302,165 @@ var _ = Describe("GCE PD CSI Driver", func() {
...
+		Expect(err).To(BeNil(), "Could not get disk from cloud directly")
+		Expect(cloudDisk.Type).To(ContainSubstring(standardDiskType))
+		Expect(cloudDisk.Status).To(Equal(readyState))
+		Expect(cloudDisk.SizeGb).To(Equal(defaultSizeGb))
+		Expect(cloudDisk.Name).To(Equal(volName))
+
+		defer func() {
+			// Delete Disk
+			controllerClient.DeleteVolume(volID)
+			Expect(err).To(BeNil(), "DeleteVolume failed")
```
Comment (msau42) [inline test/e2e/tests/single_zone_e2e_test.go]: Where is err coming from?

File: test/e2e/utils/utils.go [hunk 4]
```diff
@@ -128,6 +129,23 @@ func SetupProwConfig(resourceType string) (project, serviceAccount string) {
 	// [PROJECT_NUMBER]-compute@developer.gserviceaccount.com
 	serviceAccount = fmt.Sprintf("%v-compute@developer.gserviceaccount.com", resp.ProjectNumber)
 	glog.Infof("Using project %v and service account %v", project, serviceAccount)
+
+	// Set Cloud KMS permissions on compute service account
+	// TODO: Use the API to make this call instead of exec-ing gcloud command
+	computeSystemSA := fmt.Sprintf("service-%v@compute-system.iam.gserviceaccount.com", resp.ProjectNumber)
```
Comment (msau42) [inline test/e2e/utils/utils.go]: i'm not sure if this is the proper way we should be adding iam roles to project wide system accounts

File: test/e2e/tests/multi_zone_e2e_test.go [hunk 5]
```diff
@@ -143,68 +144,85 @@ var _ = Describe("GCE PD CSI Driver Multi-Zone", func() {
...
+	if err != nil {
+		return fmt.Errorf("ControllerPublishVolume failed with error for disk %v on node %v: %v", volID, instance.GetNodeID(), err)
+	}
 
 	defer func() {
-
 		// Detach Disk
-		err = client.ControllerUnpublishVolume(volID, instance.GetNodeID())
-		Expect(err).To(BeNil(), "ControllerUnpublishVolume failed with error")
+		_ = client.ControllerUnpublishVolume(volID, instance.GetNodeID())
```
Comment (msau42) [inline test/e2e/tests/multi_zone_e2e_test.go]: why do we not check for err?

File: test/e2e/tests/single_zone_e2e_test.go (hunk 1 above)
Comment (davidz627) [reply test/e2e/tests/single_zone_e2e_test.go]: no. I guess it should be a hard failure. this code exists from before I was creating UUID's or cleaning up crypto keys

File: test/e2e/tests/single_zone_e2e_test.go (hunk 2 above)
Comment (davidz627) [reply test/e2e/tests/single_zone_e2e_test.go]: key ring resource names are immutable.. all the tests will share the same key ring and the key ring will exist for the forseeable future. Talked to the CMEK TL and this is intended behavior

File: test/e2e/tests/single_zone_e2e_test.go (hunk 3 above)
Comment (davidz627) [reply test/e2e/tests/single_zone_e2e_test.go]: previous line but it got lost somehow

File: test/e2e/utils/utils.go (hunk 4 above)
Comment (davidz627) [reply test/e2e/utils/utils.go]: agreed, looking into alternatives now

File: test/e2e/tests/multi_zone_e2e_test.go (hunk 5 above)
Comment (davidz627) [reply test/e2e/tests/multi_zone_e2e_test.go]: fixed

File: test/e2e/tests/single_zone_e2e_test.go [hunk 6]
```diff
@@ -294,6 +302,163 @@ var _ = Describe("GCE PD CSI Driver", func() {
...
+				break
+			}
+			Expect(err).To(BeNil(), "Failed to list crypto key versions")
+
+			keyVersions = append(keyVersions, keyVersion.Name)
+		}
+
+		// Defer deletion of all key versions
+		// https://cloud.google.com/kms/docs/destroy-restore
+		// Temporarily disable revokation of CMEK key because of test infra CI permissions
```
Comment (msau42) [inline test/e2e/tests/single_zone_e2e_test.go]: this is still not resolved? will we be leaking resources?

File: test/run-e2e-local.sh [hunk 7]
```diff
@@ -5,4 +5,4 @@ set -o errexit
 
 readonly PKGDIR=sigs.k8s.io/gcp-compute-persistent-disk-csi-driver
 
-ginkgo -v "test/e2e/tests" --logtostderr -- --project ${PROJECT} --service-account ${IAM_NAME} --v=4
\ No newline at end of file
+ginkgo -v -focus=CMEK "test/e2e/tests" --logtostderr -- --project ${PROJECT} --service-account ${IAM_NAME} --v=4
```
Comment (msau42) [inline test/run-e2e-local.sh]: is this necessary to keep?

File: test/e2e/tests/single_zone_e2e_test.go (hunk 6 above)
Comment (davidz627) [reply test/e2e/tests/single_zone_e2e_test.go]: I'll add the permission for this. we are already leaking keyrings by design anyway (keyrings don't support deletion).

File: test/run-e2e-local.sh (hunk 7 above)
Comment (davidz627) [reply test/run-e2e-local.sh]: accidental `git add`

**Pull request merged**